app/components/workspace/workspace-selector/workspace-dropdown.component.ts

```diff
@@ -0,0 +1,55 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, OnDestroy } from "@angular/core";
+import { Workspace } from "app/models";
+import { WorkspaceService } from "app/services";
+import { List } from "immutable";
+import { Observable, Subscription } from "rxjs";
+
+import "./workspace-dropdown.scss";
+
+@Component({
+    selector: "bl-workspace-dropdown",
+    templateUrl: "workspace-dropdown.html",
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class WorkspaceDropDownComponent implements OnDestroy {
+    public workspaces: Observable<List<Workspace>>;
+    public selectedWorkspaceId: string = "";
+    public loaded = false;
+
+    private _subscriptions: Subscription[] = [];
+
+    constructor(
+        private changeDetector: ChangeDetectorRef,
+        private workspaceService: WorkspaceService) {
+
+        this.workspaces = this.workspaceService.workspaces;
+        this._subscriptions.push(this.workspaces.subscribe((workspaces) => {
+            console.log("workspaces.subscribe: ", workspaces);
+            this.loaded = true;
+            this.changeDetector.markForCheck();
+        }));
+
+        this._subscriptions.push(workspaceService.currentWorkspace.subscribe((workspace: Workspace) => {
```

Comment:
you probably don't need this one here. This was for accounts

Reply:
yeah this was just to test that the current workspace was being set and subscribable. will remove.